Significant-Gravitas/Auto-GPT
# Added default actions functionality

Fixes #236
Related to #150


### Background
This pull request introduces a new functionality to the `Agent` class, allowing a default number of authorizations to be set for the `start_interaction_loop()` method. During our work, we also addressed an issue with `next_action_count`, which was being treated as a string instead of an integer, causing a `TypeError` when attempting to decrement its value.

### Changes
- Implemented a new `default_authorisation_count` parameter in the `Agent` class.
- Updated the `__init__` method of the `Agent` class to cast `next_action_count` to an integer.
- Ensured that `default_authorisation_count` is also cast to an integer when updating `next_action_count`.

### Documentation
The changes made in this pull request have been documented through in-code comments and this PR message.

### Test Plan
- Run the application with various inputs to ensure the TypeError does not occur anymore.
- Verify that `next_action_count` is correctly updated and decremented as intended during the interaction loop.
- Test the new `default_authorisation_count` functionality to ensure it works as intended.

### PR Quality Checklist
